src/rgw/driver/rados/rgw_trim_bilog.cc

```diff
@@ -759,14 +760,25 @@ int BucketTrimInstanceCR::operate(const DoutPrefixProvider *dpp)
 			    << cpp_strerror(retcode) << dendl;
 	  return set_cr_error(retcode);
 	}
+
+  //remove bucket instance metadata
+  if (clean_info->first.layout.logs.front().layout.type == rgw::BucketLogType::Deleted) {
+  retcode = store->ctl()->bucket->remove_bucket_instance_info(bucket, clean_info->first, null_yield, dpp);
```

Comment:
since this is in the `BucketTrimInstanceCR` coroutine, we can't call blocking functions here. `RGWGetBucketInstanceInfoCR` and `RGWPutBucketInstanceInfoCR` are used above to read/write this metadata - it looks like we need a corresponding `RGWRemoveBucketInstanceInfoCR`

Reply:
done